benchmarks/src/has_single_bit.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <assert.h>
+#include <benchmark/benchmark.h>
+#include <bit>
+
+#include <utility.hpp>
+
+template <typename T>
+static void has_single_bit_if(benchmark::State& state) {
+    const auto random_v = random_vector<T>(8);
+    for (auto _ : state) {
+        benchmark::DoNotOptimize(random_v);
+        int count_true = 0, count_false = 0;
+        for (const auto& x : random_v) {
+            if (std::has_single_bit(x)) {
+                benchmark::DoNotOptimize(count_true++);

```

Comment:
I suspect that the signed counters might overflow for long benchmark run (and the benchmark run may be longer than usual using command-line `--benchamrk_min_time` argument). The signed overflow is Undefined Behavior. 

This can be easily avoided by using `unsinged int` instead of `int`. Unsigned overflow is defined.